packages/components/src/components/cookies-consent.tsx

```diff
@@ -1,30 +1,37 @@
 'use client';
 
-import { ComponentProps, useState } from 'react';
+import { ComponentProps, useLayoutEffect, useState } from 'react';
 import { cn } from '../cn';
 import { CallToAction } from './call-to-action';
 
 export type CookiesConsentProps = ComponentProps<'div'>;
 
 export function CookiesConsent(props: CookiesConsentProps) {
-  const [consented, setConsented] = useState(() => localStorage.getItem('cookies') === 'true');
+  const [consented, setConsented] = useState<'unknown' | 'yes' | 'no' | 'closing'>('unknown');
 
-  const onAccept = () => {
-    setConsented(true);
-    localStorage.setItem('cookies', 'true');
-  };
+  useLayoutEffect(() => {
+    if (typeof localStorage !== 'undefined') {
+      setConsented(localStorage.getItem('cookies') === 'true' ? 'yes' : 'no');
+    }
```

Comment:
```suggestion
      setConsented(localStorage.getItem('cookies') === 'true' ? 'yes' : 'no');
```
this will be not executed on server anyway